test/unit/executionPlanWebviewController.test.ts

```diff
@@ -0,0 +1,554 @@
+
+        assert.notEqual(result, undefined);
+        assert.deepStrictEqual(
+            result.executionPlanState.loadState,
+            ApiStatus.Error,
+        );
+        assert.deepStrictEqual(
+            result.executionPlanState.errorMessage,
+            "Mock Error",
+        );
+    });
+
+    test("updateTotalCost: should call updateTotalCost with the added cost and return the updated state", async () => {
+        const mockState = {
+            executionPlanState: {
+                loadState: ApiStatus.Loading,
+                executionPlanGraphs: [],
+                totalCost: 0,
+            },
+        };
+
+        const mockPayload = { addedCost: 100 };
+
+        const result = await epUtils.updateTotalCost(mockState, mockPayload);
+
+        assert.strictEqual(
+            result.executionPlanState.totalCost,
+            100,
+            "The state should be returned with new cost.",
+        );
+    });
+
+    test("calculateTotalCost: should return 0 and set loadState to Error if executionPlanGraphs is undefined", () => {
+        const mockState: any = {
+            executionPlanState: {
+                executionPlanGraphs: undefined,
+                loadState: ApiStatus.Loading,
+            },
+        };
+
+        const result = epUtils.calculateTotalCost(mockState);
+
+        assert.strictEqual(
+            result,
+            0,
+            "Total cost should be 0 when executionPlanGraphs is undefined",
+        );
+        assert.strictEqual(
+            mockState.executionPlanState.loadState,
+            ApiStatus.Error,
+            "loadState should be set to Error",
+        );
+    });
+
+    test("calculateTotalCost: should correctly calculate the total cost for a valid state", () => {
+        const mockState: any = {
+            executionPlanState: {
+                executionPlanGraphs: [
+                    { root: { cost: 10, subTreeCost: 20 } },
+                    { root: { cost: 5, subTreeCost: 15 } },
+                ],
+                loadState: ApiStatus.Loaded,
+            },
+        };
+
+        const result = epUtils.calculateTotalCost(mockState);
+
+        assert.strictEqual(
+            result,
+            50,
+            "Total cost should correctly sum up the costs and subtree costs",
+        );
+    });
+
+    test("calculateTotalCost: should return 0 if executionPlanGraphs is empty", () => {
+        const mockState: any = {
+            executionPlanState: {
+                executionPlanGraphs: [],
+                loadState: ApiStatus.Loaded,
+            },
+        };
+
+        const result = epUtils.calculateTotalCost(mockState);
+
+        assert.strictEqual(
+            result,
+            0,
+            "Total cost should be 0 for an empty executionPlanGraphs array",
+        );
+    });
+
+    test("formatXml: should return original xml contents if it is not a valid xml file", () => {
+        const invalidXml = "</";

```

Comment:
What happens when I attempt to open a sqlplan file with invalid XML?  Does it display an error message to the user?